File: app/javascript/channels/room_channel.js
import consumer from './consumer'

export function subscribeToRoom(roomId, onMessage) {
  const subscription = consumer.subscriptions.create({ channel: 'RoomChannel', room: roomId }, {
    received(data) {
      if (onMessage) onMessage(data)
    }
  })

  return subscription
}

File: app/channels/room_channel.rb
class RoomChannel < ApplicationCable::Channel
  # params: { room: room_id }
  def subscribed
    return reject unless params[:room].present?
    stream_from "room_#{params[:room]}"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end
end
